src/QsCompiler/Tests.Compiler/TestCases/QirTests/TestReferenceCounts3.ll

```diff
@@ -0,0 +1,32 @@
+define internal void @Microsoft__Quantum__Testing__QIR__Main__body() {
+entry:
+  %0 = call { %String*, %Array* }* @Microsoft__Quantum__Testing__QIR__TestPendingRefCountIncreases__body(i1 true)
+  call void @Microsoft__Quantum__Testing__QIR__TestRefCountsForItemUpdate__body(i1 true)
+  %id = call %Callable* @__quantum__rt__callable_create([4 x void (%Tuple*, %Tuple*, %Tuple*)*]* @Microsoft__Quantum__Testing__QIR_____GUID___Identity, [2 x void (%Tuple*, i32)*]* null, %Tuple* null)
+  call void @__quantum__rt__capture_update_alias_count(%Callable* %id, i32 1)
```

Comment:
One thing I'm still confused by here: why does the `%id` callable have it's capture alias and callable alias incremented here? Presumably it starts at one, then gets incremented here to two, then it is used, and then on lines 21-22 and 27-28 it is decremented twice to put it to zero and get released. It doesn't appear to be leaked at all, but it's not clear to me why it needed alias counts of 2 instead of just using the 1 it starts with and then decrementing once at the end of the scope.

Reply:
No, the alias always starts at 0. It gets incremented to 1 due to the assignment. This could potentially be optimized away, but I won't to any more involved optimizations until we have a better test coverage across the board.